jerry-core/parser/js/js-parser-statm.c

```diff
@@ -730,7 +730,10 @@ parser_parse_function_statement (parser_context_t *context_p) /**< context */
     status_flags |= PARSER_HAS_NON_STRICT_ARG;
   }
 
-  JERRY_ASSERT (context_p->next_scanner_info_p->type == SCANNER_TYPE_FUNCTION);
+  if (context_p->next_scanner_info_p->type != SCANNER_TYPE_FUNCTION)
```

Comment:
Which test triggers this? At this point the `(` should be parsed where the scanner info is attached to.

Reply:
Ok this is early. Perhaps simply removing this assertion is enough.

Reply:
This will be triggered by `function a"b`. The scanner will call `lexer_next_token` to consume the left paren, however the unterminated string literal will throw. Since the scanner info would be created after consuming the opening paren, there will be no scanner info for the function.

Reply:
Exactly. I think this assert is unnecessary, since no info is created if there will be a parse error before the `(`. I think there is another assert later which should cover this as well.